### PR TITLE
doc/dev: Structural changes to Life of a repository for readability

### PR DESCRIPTION
**Pull request by indradhanush**
Google's technical writing class suggests to:

* Use lists where possible: https://developers.google.com/tech-writing/one/lists-and-tables
*

### CODE REVIEW

**Review by unknwon — APPROVED** [review]

File: doc/dev/background-information/architecture/life-of-a-repository.md [hunk 1]
```diff
@@ -18,12 +18,19 @@ Our guiding principle is to ensure all repositories configured by a site adminis
 
 ## Services
 
-`repo-updater` is responsible for communicating with code host APIs and co-ordinating the state we synchronize from them. It is a singleton service. It is responsible for maintaining the `repo` table which other services read. It is also responsible for scheduling clones/fetches on `gitserver`. It is also responsible for anything which communicates with a code host API. So our batch changes and background permissions syncers also live in `repo-updater`.
+`repo-updater` is a singleton service. It is responsible for the following:
```
Comment (quinnkeast) [inline doc/dev/background-information/architecture/life-of-a-repository.md:21]:
```suggestion
`repo-updater` is a singleton service. It is responsible for:
```

File: doc/dev/background-information/architecture/life-of-a-repository.md [hunk 2]
```diff
@@ -65,12 +72,12 @@ The scheduler is divided into two parts:
 - [`updateQueue`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@v3.14.0/-/blob/cmd/repo-updater/repos/scheduler.go#L392:6) is a priority queue of repositories to clone/fetch on `gitserver`.
 - [`schedule`](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@v3.14.0/-/blob/cmd/repo-updater/repos/scheduler.go#L567:6) which places repositories onto the `updateQueue` when it thinks it should be updated. This is what paces out updates for a repository. It contains heuristics such that recently updated repositories are more frequently checked.
 
-Repositories can also placed onto the `updateQueue` if we receive a webhook indicating the repository has changed. (We don't by default setup webhooks when integrating into a code host). When a user directly visits a repository on Sourcegraph we also enqueue it for update.
+Repositories can also be placed onto the `updateQueue` if we receive a webhook indicating the repository has changed. (By default we don't setup webhooks when integrating into a code host.) When a user directly visits a repository on Sourcegraph we also enqueue it for update.
```
Comment (quinnkeast) [inline doc/dev/background-information/architecture/life-of-a-repository.md:75]:
```suggestion
Repositories can also be placed onto the `updateQueue` if we receive a webhook indicating the repository has changed. (By default, we don't set up webhooks when integrating into a code host.) When a user directly visits a repository on Sourcegraph, we also enqueue it for update.
```

File: doc/dev/background-information/architecture/life-of-a-repository.md [hunk 3]
```diff
@@ -18,12 +18,19 @@ Our guiding principle is to ensure all repositories configured by a site adminis
...
+`repo-updater` is a singleton service. It is responsible for the following:
 
-`gitserver` is a scaleable stateful service which clones git repositories and can run git commands against them. All data maintained on this service is from cloning an upstream repository. We shard the set of repositories across the gitserver replicas. The main RPC gitserver supports is `exec` which returns the output of the specified git command.
+* Communicating with code host APIs to coordinate the state we synchronize from them.
+* Maintaining the `repo` table which other services read.
+* Scheduling clones/fetches on `gitserver`.
+* Anything which communicates with a code host API.
+
+Our batch changes and background permissions syncers also live in `repo-updater` as they require communication with code host APIs.
```
Comment (quinnkeast) [inline doc/dev/background-information/architecture/life-of-a-repository.md:28]:
```suggestion
Our batch changes and background permissions syncers are also located in `repo-updater` as they require communication with code host APIs.
```

Comment (quinnkeast) [reply doc/dev/background-information/architecture/life-of-a-repository.md:28]: This one I could go either way on. "Live in" is an idiom that's better avoided, but it's also consistently used by developers... your call!

**Review by quinnkeast — APPROVED** [review]

**Comment by quinnkeast** [comment]
> Google's technical writing class suggests to:
> 
> * Use lists where possible: https://developers.google.com/tech-writing/one/lists-and-tables

Great changes!

It might be worth including some of your learnings in [our own content guidelines](https://about.sourcegraph.com/handbook/communication/content_guidelines), specific for documentation (versus product copy).

**Comment by indradhanush** [comment]
> It might be worth including some of your learnings in our own content guidelines, specific for documentation (versus product copy).

Noted. I'll take a look at this to see what could be improved there!

**Pull request merged**